File: my-code/278/main.go
package main

import "fmt"

// @Time     : 2021/6/13 08:49
// @Author   : Ranshi
// @File     : 278. 第一个错误的版本.go
func isBadVersion(version int) bool {
	return true
}

func firstBadVersion(n int) int {
	l, r := 1, n
	for l < r {
		mid := (r + l) / 2
		fmt.Println(mid)
		if isBadVersion(mid) {
			r = mid - 1
		} else {
			l = mid + 1
		}
	}
	if isBadVersion(l) {
		return l
	} else {
		return l + 1
	}
}

File: my-code/278/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2021/08/26 17:40
# @Author   : Ranshi
# @File     : 123.py


class Solution:
    def firstBadVersion(self, n: int) -> int:
        le, ri = 0, n
        while le < ri:
            idx = (le + ri) // 2
            if isBadVersion(idx):
                ri = idx
            else:
                le = idx + 1
        return le
